dependency-lint.yml

```diff
@@ -19,7 +19,8 @@ executedModules:
 ignoreErrors:
   missing: []
   shouldBeDependency: []
-  shouldBeDevDependency: []
+  shouldBeDevDependency:
+    - tmp # used in http_stream.ts as well as in test code
```

Comment:
If this is now used in source code we should move this from a dev dependency to a dependency and remove the need to ignore the errory

Reply:
Removed as appears unnecessary since tmp is already moved to a dependency

Reply:
Resolving this since @charlierudolph fixed it in 841d8d6b4ffef2f5a4917fd97e655f4d4a575258